geovin/sd-v3/src/index.ts

```diff
@@ -0,0 +1,1167 @@
+            id: "myViewport",
+            branding: {
+                backgroundColor: "#ffffff",
+            }
+        });
+        this.currentViewport = viewport;
+        
+    }
+
+    public async adjustPeripherals() {
+        console.log('adjusting peripherals',this.isBed,this.isPainted);
+        if (this.activeShape === "G08" || this.activeShape === "G09" || this.activeShape === "G19") { //need to figure this out for G10, G11 also adjust value for light scene
+            //this.rendering = true;
+            const parameterFMetal = await this.currentSession.getParameterByName('Fabric-Metal')[0];
+            const parameterMMetal = await this.currentSession.getParameterByName('Mattress-Metal')[0];
+            //this isn't resolving and leaving us in rendering state
+            if ( this.isPainted ) {
+                parameterFMetal.value = 0.65;
+                parameterMMetal.value = 0.65;
+                await this.currentSession.customize()
+            } else {
+                parameterFMetal.value = 0.35;
+                parameterMMetal.value = 0.35;
+                await this.currentSession.customize()
+            }
+            
+            
+        }
+        if (this.activeScene != 'Scene Geovin') {
+            this.rendering = true;
+            this.setLightingScene( 'Scene Geovin' );
+        }
+        /*
+        if (this.isPainted) {
+            
+        } else {
+            if (this.activeScene != 'Scene MAW') {
+                this.rendering = true;
+                this.setLightingScene( 'Scene MAW' );
+            }
+        }*/
+    }
+
+//resolve metalness for beds that have it
+    private async setLightingScene( scene: string ) {
+        console.log('setting scene',scene );
+        await this.currentViewport.assignLightScene( scene );
+        this.activeScene = scene;
+        /*
+        await Object.values(this.availableScenes).forEach( async (value:any) => {
+            if (value.name === scene) {
+                console.log('assigning scene',value);
+                await this.currentViewport.assignLightScene( value.name );
+                this.activeScene = value.name;
+                //const lightScene = this.currentViewport.lightScene!;
+                //console.log('lightscene',lightScene);
+                //lightScene.environmentMap = "default";
+                //lightScene.environmentMapIntensity = 0;
+                //lightScene.environmentMapAsBackground = true;
+            }
+        } );*/
+    }
+
+    private async adjustZoom() {
+        const camera = this.currentViewport.camera!;
+        camera.autoAdjust = true;
+        camera.zoomToFactor = 1.4;
+        camera.enableZoom = false;
+        await this.currentViewport.render();
+    }
+
+    private async enableZoom() {
+        const camera = this.currentViewport.camera!;
+        camera.enableZoom = true;
+        await this.currentViewport.render();
+    }
+
+    private async adjustCamera() {
```

Comment:
I like the way these functions are broken down, taking advantage of multiple threads being able to fire off at once.